test/e2e_yaml_test.go

```diff
@@ -0,0 +1,238 @@
+// +build e2e,yaml
```

Comment:
this is a total nit but can we change "yaml" to "examples"? i think "example tests" is a bit more descriptive than "yaml tests" :D

Reply:
(and the same for the file name and function names that refer to Yaml) 🙏